dozer/cogs/starboard.py

```diff
@@ -0,0 +1,366 @@
+import logging
+import typing
+
+import discord
+from discord.ext.commands import guild_only, has_permissions
+
+from ._utils import *
+from .. import db
+
+MAX_EMBED = 1024
+DOZER_LOGGER = logging.getLogger('dozer')
+VIDEO_FORMATS = ['.mp4', '.mov', 'webm']
+
+
+async def is_cancelled(config, message, me, author=None):
+    """Determine if the message has cancellation reacts"""
+    if author is None:
+        author = message.author
+
+    for reaction in message.reactions:
+        if str(reaction) != config.cancel_emoji:
+            continue
+
+        users = await reaction.users().flatten()
+        if author in users or me in users:
+            await message.add_reaction(config.cancel_emoji)
+            return True
+
+    return False
+
+
+def make_starboard_embed(msg: discord.Message, reaction_count):
+    """Makes a starboard embed."""
+    e = discord.Embed(color=msg.author.color, title="New Starred Message")
+    e.set_author(name=msg.author.display_name, icon_url=msg.author.avatar_url)
+
+    view_link = f" [[view]]({msg.jump_url})"
+    if not len(msg.content):
+        e.add_field(name="Link:", value=view_link)
+    elif len(msg.content) < (MAX_EMBED - len(view_link)):
+        e.add_field(name="Content:", value=msg.content + view_link)
+    elif len(msg.content) < MAX_EMBED:
+        e.add_field(name="Content:", value=msg.content)
+        e.add_field(name="Link:", value=view_link)
+    elif len(msg.content) < ((2*MAX_EMBED) - len(view_link)):
+        e.add_field(name="Content:", value=msg.content[0:MAX_EMBED], inline=False)
+        e.add_field(name="More:", value=msg.content[MAX_EMBED:(2*MAX_EMBED) - len(view_link)] + view_link, inline=False)
+    else:
+        e.add_field(name="Content:", value=msg.content[0:MAX_EMBED], inline=False)
+        e.add_field(name="More:", value=msg.content[MAX_EMBED:2*MAX_EMBED], inline=False)
+        e.add_field(name="Link:", value=view_link)
+
+    if len(msg.attachments) > 1:
+        e.add_field(name="Attachments:", value="\n".join([f"[{a.filename}]({a.url})" for a in msg.attachments]))
+    elif len(msg.attachments) == 1:
+        if msg.attachments[0].width is not None and msg.attachments[0].filename[-4:] not in VIDEO_FORMATS:
+            e.set_image(url=msg.attachments[0].url)
+        else:
+            e.add_field(name="Attachment:", value=f"[{msg.attachments[0].filename}]({msg.attachments[0].url})")
+
+    e.set_footer(text=f"{reaction_count} reactions")
+    e.timestamp = msg.created_at
+    return e
+
+
+class Starboard(Cog):
+    """Cog to post specific 'Hall of Fame' messages in a specific channel"""
+
+    def __init__(self, bot):
+        super().__init__(bot)
+        self.config_cache = {}
+
+    def make_config_embed(self, ctx, title, config):
+        """Makes a config embed."""
+        e = discord.Embed(title=title, color=discord.Color.gold())
+        e.add_field(name="Starboard Channel", value=self.bot.get_channel(config.channel_id).mention)
```

Comment:
This will fail if the starboard channel has been deleted. It would be good to show that the channel needs to be reconfigured for the starboard to continue to work.